src/LanguageServer/Protocol/Handler/DataTips/DataTipRangeHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Debugging;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler;
+
+[ExportCSharpVisualBasicStatelessLspService(typeof(DataTipRangeHandler)), Shared]
+[Method(VSInternalMethods.TextDocumentDataTipRangeName)]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class DataTipRangeHandler()
+    : ILspServiceDocumentRequestHandler<TextDocumentPositionParams, VSInternalDataTip?>
+{
+    public bool MutatesSolutionState => false;
+    public bool RequiresLSPSolution => true;
+
+    public TextDocumentIdentifier GetTextDocumentIdentifier(TextDocumentPositionParams request)
+        => request.TextDocument;
+
+    public async Task<VSInternalDataTip?> HandleRequestAsync(TextDocumentPositionParams request, RequestContext context, CancellationToken cancellationToken)
+    {
+        var document = context.GetRequiredDocument();
+
+        var service = document.GetLanguageService<ILanguageDebugInfoService>();
+        if (service is null)
```

Comment:
Can this ever be null?  I think I saw a C#/VB/F# implementation

Reply:
i wasn't certain what our rules were with LSP.  like how xaml works there.  i wwanted to be a little safe.

Reply:
this is exported as `ExportCSharpVisualBasicStatelessLspService`, so it applies only to the main roslyn server (which supports C#,VB, and F#).  It won't be called for anything else